app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkTouchHelper.kt

```diff
@@ -41,7 +42,9 @@ class BookmarkTouchCallback(private val interactor: BookmarkViewInteractor) :
      */
     override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
         val item = (viewHolder as BookmarkNodeViewHolder).item
-        item?.let { interactor.onDelete(setOf(it)) }
+        item?.let {
```

Comment:
Based on the previous comments which show that we cannot really obtain an MVI loop for this change I propose going with your first implementation (https://github.com/person808/fenix/commit/55665f13fdd979aae5d8f7a12abe88d556f4f144) here, just that instead of having the adapter being passed to BookmarkTouchHelper we could simply use `viewHolder.bindingAdapter?.notifyItemChanged(viewHolder.bindingAdapterPosition)`.
What do you say?

Reply:
Maybe not the best solution.. but if the touch helper knows to swipe the item.. it could also know to revert that..?